File: TCS/Practice/neighbours.py
array = input("Enter the elements of the array separated by spaces: ")
array = list(map(int, array.split()))

# Variable to store the last peak element found
last_peak = None

# Check for peak elements
for i in range(1, len(array) - 1):
    if array[i] >= array[i - 1] and array[i] >= array[i + 1]:
        last_peak = array[i]

# Check the edge cases
if len(array) > 1:
    if array[0] >= array[1]:
        last_peak = array[0]
    if array[-1] >= array[-2]:
        last_peak = array[-1]

if last_peak is not None:
    print("Peak element:", last_peak)
else:
    print("No peak element found")

File: TCS/Practice/neighbours.cpp
#include <iostream>

using namespace std;

int main() {
    const int maxSize = 100;
    int array[maxSize];
    int n;

    cout << "Enter the number of elements in the array: ";
    cin >> n;

    if (n <= 0) {
        cout << "No peak element found" << endl;
        return 0;
    }

    cout << "Enter the elements of the array: ";
    for (int i = 0; i < n; ++i) {
        cin >> array[i];
    }

    // Check the first element
    if (array[0] >= array[1]) {
        cout << "Peak element: " << array[0] << endl;
        return 0;
    }

    // Check the last element
    if (array[n - 1] >= array[n - 2]) {
        cout << "Peak element: " << array[n - 1] << endl;
        return 0;
    }

    // Check the middle elements
    for (int i = 1; i < n - 1; ++i) {
        if (array[i] >= array[i - 1] && array[i] >= array[i + 1]) {
            cout << "Peak element: " << array[i] << endl;
            return 0;
        }
    }

    cout << "No peak element found" << endl;
    return 0;
}
